sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/EventBatchContext.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs.models;
+
+import com.azure.messaging.eventhubs.CheckpointStore;
+import com.azure.messaging.eventhubs.EventData;
+import com.azure.messaging.eventhubs.EventProcessorClientBuilder;
+import java.util.List;
+import java.util.function.Consumer;
+import reactor.core.publisher.Mono;
+
+/**
+ * A class that contains a batch of {@link EventData} and the partition information the event batch belongs to. This is
+ * given to the {@link EventProcessorClientBuilder#processEventBatch(Consumer, int)}  processEventBatch} handler each
+ * time an event batch is received from the Event Hub. This class also includes methods to update checkpoint in
+ * {@link CheckpointStore} and retrieve the last enqueued event information.
+ */
+public class EventBatchContext {
+
+    private final PartitionContext partitionContext;
+    private final List<EventData> eventDataList;
+    private final CheckpointStore checkpointStore;
+    private final LastEnqueuedEventProperties lastEnqueuedEventProperties;
+
+    /**
+     * Creates an instance of {@link EventContext}.
+     *
+     * @param partitionContext The partition information associated with the received event.
+     * @param eventDataList The list of events received from Event Hub.
+     * @param checkpointStore The checkpoint store that is used for updating checkpoints.
+     * @param lastEnqueuedEventProperties The properties of the last enqueued event in this partition. If {@link
+     * EventProcessorClientBuilder#trackLastEnqueuedEventProperties(boolean)} is set to {@code false}, this will be
+     * {@code null}.
+     * @throws NullPointerException If {@code partitionContext}, {@code eventData} or {@code checkpointStore} is null.
+     */
+    public EventBatchContext(PartitionContext partitionContext, List<EventData> eventDataList,
+        CheckpointStore checkpointStore, LastEnqueuedEventProperties lastEnqueuedEventProperties) {
+        this.checkpointStore = checkpointStore;
+        this.eventDataList = eventDataList;
+        this.lastEnqueuedEventProperties = lastEnqueuedEventProperties;
+        this.partitionContext = partitionContext;
+    }
+
+    /**
+     * Returns the partition information associated with the received event.
+     *
+     * @return The partition information of the received event.
+     */
+    public PartitionContext getPartitionContext() {
+        return partitionContext;
+    }
+
+    /**
+     * Returns a list of event data received from Event Hub.
+     *
+     * @return The list of event data received from Event Hub.
+     */
+    public List<EventData> getEventDataList() {
```

Comment:
I may have tapped out during this API review, but `getEvents()` seems less cumbersome than getEventDataList.

Reply:
I agree

Reply:
Updated name to `getEvents()`